server/paymentProviders/paypal/payment.js

```diff
@@ -186,12 +186,21 @@ export const refundPaypalCapture = async (transaction, captureId, user, reason)
     throw new Error(`PayPal: Can't find host for transaction #${transaction.id}`);
   }
 
-  // eslint-disable-next-line camelcase
-  const payload = { note_to_payer: truncate(reason, { length: 255 }) || undefined };
-  const result = await paypalRequestV2(`payments/captures/${captureId}/refund`, host, 'POST', payload);
-  const rawRefundedPaypalFee = get(result, 'seller_payable_breakdown.paypal_fee.amount.value', '0.00');
-  const refundedPaypalFee = floatAmountToCents(parseFloat(rawRefundedPaypalFee));
-  return createRefundTransaction(transaction, refundedPaypalFee, { paypalResponse: result }, user);
+  // Add a flag on transaction to prevent multiple calls from the webhook
+  await transaction.update({ data: { ...transaction.data, isRefundedFromOurSystem: true } });
```

Comment:
This is smart, we already had echoes from Paypal before.
If the early return lives in the `webhook.ts:handleCaptureRefunded` I don't really see the benefit of having this lock/try/catch logic here. I think it would be simpler if we had the whole lock logic inside `handleCaptureRefunded`.

Reply:
The main idea here is to ignore the webhook event if the transaction is refunded from Open Collective's interface (because the `refundTransaction` mutation already takes care of everything). It's not intended to prevent PayPal echoes, that part is taken care of in the webhook itself  (by making sure `RefundTransactionId: null`).

But I realize that the comment is ambiguous, I'm going to push an update for it